src/arrayMethodJoin.js

```diff
@@ -5,7 +5,33 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    // write code her
+    let devider = separator;
+
+    if (devider === undefined) {
+      devider = ',';
+    }
+
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === (this.length - 1)) {
+        if (this[i] === undefined) {
+          break;
+        }
+        result = result + this[i];
+        break;
+      }
+
+      if (this[i] === undefined || this[i] === null) {
+        result = result + devider;
+        continue;
+      }
+
+      result = result + this[i] + devider;
```

Comment:
You can simplify it
if element is not undefined or null concate it with result then if element is not a last concate separator. That's all what you need to do